### PR TITLE
Disable "warning as error" option in "Ship" configuration on windows

### PR DESCRIPTION
**Pull request by Blackhart**
related issue: #2272

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -127,7 +127,12 @@ option (USE_STATIC_OCIO                     "Use static OpenColorIO libraries"
 option (USE_STATIC_OIIO                     "Use static OpenImageIO libraries"                          ON)
 option (USE_STATIC_OSL                      "Use static OpenShadingLanguage libraries"                  ON)
 
-option (WARNINGS_AS_ERRORS                  "Treat compiler warnings as errors"                         ON)
+if (CMAKE_SYSTEM_NAME STREQUAL "Windows" AND CMAKE_BUILD_TYPE MATCHES "Ship")
```
[the remark this answers is not in the document]
Comment (Blackhart) [reply CMakeLists.txt]: I know that ...
I'm idiot!

**Comment by Blackhart** [comment]
I updated the windows specific cmake options file.
There is one "warning as error" checking for each build type now. 
Except for "Ship" build type.

I think it's more redundant to check it for each build types.
Is it better to check it at the end of build type configurations and set it in one go ?

**Comment by dictoon** [comment]
You're right, let's do it like it's done on Linux and macOS, in the `Flags common to all configurations` section:
```
if (WARNINGS_AS_ERRORS)
    if (CMAKE_BUILD_TYPE STREQUAL "Debug" OR
        CMAKE_BUILD_TYPE STREQUAL "Profile" OR
        CMAKE_BUILD_TYPE STREQUAL "Release")
        set (c_compiler_flags_common
            ${c_compiler_flags_common}
            -Werror                                     # treat Warnings As Errors
        )
    endif ()
endif ()
```

(Make sure to check how it's actually done in `linux-gcc.txt`.)

While we're at it: Can we also disable warnings-as-errors for Profile builds? We just keep them for Debug and Release builds.

**Comment by dictoon** [comment]
Superseded by https://github.com/appleseedhq/appleseed/pull/2406.

**Pull request closed without merging**